html/controls/plt/plt-input.js

```diff
@@ -442,17 +455,28 @@ function preparePenaltyEditor() {
 
 	function initialize() {
 		
-		var topTable = $('<table width="80%">').appendTo($('#PenaltyEditor'));
+		var topTable = $('<table width="100%">').appendTo($('#PenaltyEditor'));
 		var tr = $('<tr>').appendTo(topTable);
-		$('<td width="33%">').append($('<span>').text('Period: ')).append($('<button>').addClass('period_minus').text('-1'))
-			.append($('<select>').addClass('Period')).append($('<button>').addClass('period_plus').text('+1')).appendTo(tr);
-		$('<td width="33%">').append($('<span>').text('Jam: ')).append($('<button>').addClass('jam_minus').text('-1'))
-			.append($('<select>').addClass('Jam')).append($('<button>').addClass('jam_plus').text('+1')).appendTo(tr);
-		$('<td width="33%">').append($('<button>').addClass('clear').text('Clear')).appendTo(tr);
+		$('<td width="35%">').append($('<span>').text('Period: ')).append($('<button>').addClass('period_minus').text('-1').button())
+			.append($('<select>').addClass('Period')).append($('<button>').addClass('period_plus').text('+1').button()).appendTo(tr);
+		$('<td width="35%">').append($('<span>').text('Jam: ')).append($('<button>').addClass('jam_minus').text('-1').button())
+			.append($('<select>').addClass('Jam')).append($('<button>').addClass('jam_plus').text('+1').button()).appendTo(tr);
+		$('<td width="15%">').append($('<button>').addClass('set').text('Set Jam').button()).appendTo(tr);
+		$('<td width="15%">').append($('<button>').addClass('clear').text('Clear').button()).appendTo(tr);
 		
 		var tr2 = $('<tr>').appendTo(topTable);
-		$('<td>').append($('<input type="checkbox">').attr('id', 'served'))
-			.append($('<span>').text(' has been served')).appendTo(tr2);
+		$('<td>').append($('<button>').text('Served').attr('id', 'served').button().click(function() {
+			var active = $(this).hasClass('checked');
+			$(this).toggleClass('checked', !active);
+			if (!isTrue(penaltyEditor.data('new'))) {
+				var teamId = penaltyEditor.data('team');
+				var skaterId = penaltyEditor.data('skater');
+				var penaltyNumber = penaltyEditor.data('pnr');
+				var prefix = 'ScoreBoard.Team(' + teamId + ').Skater(' + skaterId + ').Penalty(' + penaltyNumber + ')';
+				WS.Set(prefix + '.ForceServed', active);
+				penaltyEditor.dialog('close');
```

Comment:
This shouldn't close the dialog.

We also need an explicit close button now

Reply:
When editing a penalty that already exists, it's way more likely that only one attribute is supposed to be changed. So closing on edit is serving the more common workflow.